aws/sdk/examples/telephone-game/src/main.rs

```diff
@@ -0,0 +1,432 @@
+    runtime::Runtime,
+};
+
+/// While playing the telephone game, the user can pass an arg that defines how many times to pass the message through Polly and Transcribe.
+/// This is the default number of iterations to do when the user doesn't specify.
+const DEFAULT_NUMBER_OF_ITERATIONS: u32 = 5;
+/// When running a job/task that takes some time to complete (speech synthesis and transcription), this sets a maximum wait time in seconds before giving up.
+const TASK_TIMEOUT_IN_SECONDS: i32 = 30;
+/// How often to poll for job/task status
+const TASK_WAIT_INTERVAL_IN_SECONDS: i32 = 2;
+
+fn main() {
+    // Read ENV vars from a .env file if one exists
+    // To see logs set RUST_LOG="error,telephone_game=debug"
+    dotenv::dotenv().expect("failed to read env vars");
+    // Start up the logger
+    env_logger::init();
+    let app = build_clap_app();
+
+    match app.get_matches().subcommand() {
+        ("play", Some(matches)) => play_telephone(matches),
+        ("parrot", _) => test_polly(),
+        _ => (),
+    }
+}
+
+fn build_clap_app<'a, 'b>() -> clap::App<'a, 'b> {
+    clap::App::new(crate_name!())
+        .version(crate_version!())
+        .author(crate_authors!())
+        .about(crate_description!())
+        .subcommand(
+        clap::App::new("play")
+                .about("Start playing a game of Telephone")
+                .args_from_usage("--phrase=[PHRASE] 'The phrase to play the game with'")
+                .args_from_usage("-i [iterations] 'The number of times to relay the telephone message, defaults to 1 when omitted'")
+                .args_from_usage("-b [s3_bucket_name] 'The name of the S3 bucket that will be used to store intermediate audio and text files created by the game, defaults to telephone-game when omitted'")
+        )
+        .subcommand(clap::App::new("parrot").about("hear polly repeat what you say"))
+}
+
+/// Make Polly speak what you type
+fn test_polly() {
+    // Create a runtime so we can do some async Rust
+    let rt = Runtime::new().expect("failed to create an async runtime");
+    // Create a string to store user input
+    let mut line = String::new();
+
+    rt.block_on(async {
+        // Create a new AWS Config
+        let config = aws_config::load_from_env().await;
+        let polly_client = aws_sdk_polly::Client::new(&config);
+
+        // Set up a temp directory to store audio files
+        let tmp_dir = TempDir::new("telephone-game").expect("couldn't create temp dir");
+        let tmp_file_path = tmp_dir.path().join("polly.mp3");
+
+        // Set up the ability to play audio
+        let (_stream, stream_handle) =
+            OutputStream::try_default().expect("Couldn't get handle to default audio output");
+        let sink = Sink::try_new(&stream_handle).unwrap();
+
+        loop {
+            info!(r#"Say something and I'll echo it or say "quit" to quit"#);
+            // Collect user input
+            let mut input_lines = BufReader::new(tokio::io::stdin()).lines();
```

Comment:
Should this be outside of the loop?

Reply:
I refactored this so this comment no longer applies